### PR TITLE
Add WebSocket sample

### PR DESCRIPTION
**Pull request by sebastianbenz**
This sample demonstrates how to use a websocket in a service worker.

### CODE REVIEW

File: functional-samples/tutorial.websockets/manifest.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "WebSocket Demo",
+  "description": "How to use WebSockets in your Chrome Extension.",
+  "version": "1.0",
+  "manifest_version": 3,
+  "action": {
+    "default_icon": "icons/socket-inactive.png"
+  },
+  "permissions": ["storage"],
```
Comment (AmySteam) [inline functional-samples/tutorial.websockets/manifest.json:9]: This permission is not used

File: functional-samples/tutorial.websockets/manifest.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (AmySteam) [inline functional-samples/tutorial.websockets/manifest.json:1]: Is a minimum chrome version relevant for this example?

File: functional-samples/tutorial.websockets/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Chrome Extension WebSocket Example
```
Comment (AmySteam) [inline functional-samples/tutorial.websockets/README.md:1]:
```suggestion
# Chrome extension WebSocket example
```

File: functional-samples/tutorial.websockets/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+This example demonstrates how to use WebSockets in a MV3 Chrome Extension.
+
```
Comment (AmySteam) [inline functional-samples/tutorial.websockets/README.md:4]: Do we need to add a minimum Chrome version prereq?

File: functional-samples/tutorial.websockets/icons/socket-active.png
Comment (AmySteam) [inline functional-samples/tutorial.websockets/icons/socket-active.png:1]: These icons look familiar 🧐

**Review by AmySteam — COMMENTED** [review]
Thanks for this sample @sebastianbenz 💯 

I didn't test this, but left a few fun comments and questions.

File: functional-samples/tutorial.websockets/README.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Chrome Extension WebSocket Example
+
+This example demonstrates how to use WebSockets in a MV3 Chrome Extension starting with Chrome version M116.
+
+## Running this extension
+
+1. Clone this repository.
```
Comment (oliverdunk) [inline functional-samples/tutorial.websockets/README.md:7]: From what I can tell, if the Glitch app goes to sleep you have to wake it over HTTP before you can connect on a WebSocket. Can we add a note about that (e.g visit this HTTP URL and then wait a few seconds) before trying the sample?

File: functional-samples/tutorial.websockets/README.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# Chrome Extension WebSocket Example
+
+This example demonstrates how to use WebSockets in a MV3 Chrome Extension starting with Chrome version M116.
```
Comment (oliverdunk) [inline functional-samples/tutorial.websockets/README.md:3]: Can we say that from Chrome 116 the service worker should stay alive?

**Review by oliverdunk — APPROVED** [review]
Left a couple of nits but otherwise the code looks great! Thanks for this 😁

File: functional-samples/tutorial.websockets/manifest.json (hunk 1 above)
Comment (sebastianbenz) [reply functional-samples/tutorial.websockets/manifest.json]: removed

File: functional-samples/tutorial.websockets/icons/socket-active.png
Comment (sebastianbenz) [reply functional-samples/tutorial.websockets/icons/socket-active.png:1]: :-) I was inspired...

File: functional-samples/tutorial.websockets/README.md (hunk 6 above)
Comment (sebastianbenz) [reply functional-samples/tutorial.websockets/README.md]: done

File: functional-samples/tutorial.websockets/README.md (hunk 4 above)
Comment (sebastianbenz) [reply functional-samples/tutorial.websockets/README.md:6]: good point - done!

**Review by oliverdunk — APPROVED** [review]

**Pull request merged**